src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -68,6 +69,9 @@ private MethodSpec(Builder builder) {
     this.parameters = Util.immutableList(builder.parameters);
     this.varargs = builder.varargs;
     this.exceptions = Util.immutableList(builder.exceptions);
+    this.initializer = (builder.initializer == null)
```

Comment:
I wanted to remain consistent with your initializer in FieldSpec but no problem, I can modify it!